### PR TITLE
tree-wide: CMP()ify all the things

### PR DESCRIPTION
**Pull request by poettering**
Let's employ coccinelle to fix everything up automatically for us.

### CODE REVIEW

File: src/basic/btrfs-util.c [hunk 1]
```diff
@@ -418,12 +418,7 @@ static int btrfs_ioctl_search_args_compare(const struct btrfs_ioctl_search_args
         if (args->key.min_type > args->key.max_type)
                 return 1;
```
Comment (yuwata) [inline src/basic/btrfs-util.c:419]: Here, we can also use `CMP()`...

Comment (poettering) [reply src/basic/btrfs-util.c:419]: yupp, but the cociccenelle check is harder, so i was lazy for for now... but yes, we should definitely add that later, too

**Comment by evverx** [comment]
By the way, the authors of Coccinelle seem to be gathering use cases in https://systeme.lip6.fr/pipermail/cocci/2018-October/005351.html. I suppose the most active users could write something :-)

**Comment by poettering** [comment]
> By the way, the authors of Coccinelle seem to be gathering use cases in https://systeme.lip6.fr/pipermail/cocci/2018-October/005351.html. I suppose the most active users could write something :-)

Quite frankly I am not too sold on the tool, in general. It's so clumsy in many ways, and unhackable (ocaml, ffs!?!)... And sometimes the output is not really the coding style we'd like to see.. I wish we had something smarter, maybe that actually understands what is being processed here, i.e something that maybe is integrated with the llvm AST, and can transform trees in a smart fashion. Right now our cocci hookup is mostly because it's kinda the only tool for the job, not because I love it so much.

**Comment by evverx** [comment]
Well, negative feedback is feedback too :-)

> maybe that actually understands what is being processed here

In theory, it's possible to add custom LGTM queries written in the QL language to look for common issues. Unfortunately, it can't be used to transform code automatically, but as long as PRs introducing new alerts aren't merged it doesn't seem to be critical :-)

**Pull request merged**